### ISSUE TITLE
save_model() Feature Importance Section can use a different importance_type

### ISSUE THREAD

**Issue by little-eyes** [microsoft/LightGBM#3011]
## Summary

Currently, the booster.save_model() function does not accept additional params to change the importance type when output the feature importance to the model file. The c++ layer is also hard coded to use 'split' as the importance_type as below.

https://github.com/microsoft/LightGBM/blob/9478e52f9fd751353d4e0e04899c2b090f1ff0fe/src/boosting/gbdt_model_text.cpp#L367

So it would be great to have an additional params to be able to change the behavior of feature importance, e.g. uses 'gain' as importance type.

## Motivation

We uses LightGBM in our production work, and we show some diagnostic plots, such as feature importance plot, to our customers. But when they inspect the model file, they found a totally different feature importance results than our plots because we use 'gain' as importance_type, while the model file uses 'split'. So it would be great to show consistency to our customers by using the same importance type.

## Description

Can we expose APIs like below:

```python
model.booster_.save_model(path, model.booster_.best_iteration, importance_type='gain')
```

**Comment by StrikerRUS** [comment]
Refer to https://github.com/microsoft/LightGBM/pull/2656#issuecomment-570387134 and ping @guolinke .

**Comment by StrikerRUS** [comment]
Closed via #3220.
